File: app/src/main/java/com/stavros/demo/android/form/LandlordRegistration.java
package com.stavros.demo.android.form;

import androidx.appcompat.app.AppCompatActivity;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Toast;

import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.stavros.demo.android.R;

import java.util.HashMap;
import java.util.Map;

public class LandlordRegistration extends AppCompatActivity {

    private EditText uFirstName,
            uLastName,
            uPhoneNumber,
            uID_number,
            uEmail;

    private RadioGroup radiog_Gender;
    private RadioButton uGender;
    private Button btn_save;
    private DatabaseReference landlordRef;
    private final ThreadLocal<ProgressDialog> progressDialog = new ThreadLocal<ProgressDialog>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_landlord_registration);

        uFirstName = (EditText) findViewById(R.id.firstName);
        uLastName = (EditText) findViewById(R.id.lasttName);
        uPhoneNumber = (EditText) findViewById(R.id.phoneNumber);
        uID_number = (EditText) findViewById(R.id.id_number);
        uEmail = (EditText) findViewById(R.id.email);
        radiog_Gender = (RadioGroup) findViewById(R.id.gender);
        btn_save = (Button) findViewById(R.id.btn_save);

        progressDialog.set(new ProgressDialog(this));
        btn_save.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                saveLandlord();
            }
        });
    }

    private void saveLandlord() {
        final String firstName = uFirstName.getText().toString().trim();
        final String lastName = uLastName.getText().toString().trim();
        final String phone = uPhoneNumber.getText().toString().trim();

        int genderBtn = radiog_Gender.getCheckedRadioButtonId();

        String gender = "";
        uGender = (RadioButton) findViewById(genderBtn);

        final String id_number = uID_number.getText().toString().trim();
        final String email = uEmail.getText().toString().trim();

        if (TextUtils.isEmpty(firstName)) {
            Toast.makeText(this, getResources().getString(R.string.Please_enter_first_name), Toast.LENGTH_LONG).show();
            return;
        }

        if (TextUtils.isEmpty(lastName)) {
            Toast.makeText(this, getResources().getString(R.string.Please_enter_last_name), Toast.LENGTH_LONG).show();
            return;
        }

        if (TextUtils.isEmpty(phone)) {
            Toast.makeText(this, getResources().getString(R.string.Please_enter_phone_number), Toast.LENGTH_LONG).show();
            return;
        } else if (phone.length() != 10) {
            Toast.makeText(this, getResources().getString(R.string.Please_enter_phone_number), Toast.LENGTH_LONG).show();
            return;
        }

        if (uGender != null) {
            gender = uGender.getText().toString();
        } else {
            Toast.makeText(this, getResources().getString(R.string.Please_select_gender), Toast.LENGTH_LONG).show();
            return;
        }

        if (TextUtils.isEmpty(id_number)) {
            Toast.makeText(this, getResources().getString(R.string.Please_enter_id_number), Toast.LENGTH_LONG).show();
            return;
        }

        if (TextUtils.isEmpty(email)) {
            Toast.makeText(this, getResources().getString(R.string.Please_enter_email), Toast.LENGTH_LONG).show();
            return;
        }

        landlordRef = FirebaseDatabase.getInstance().getReference().child("Users").child("Landlords").child(id_number);

        Map newPost = new HashMap();
        newPost.put("firstName", firstName);
        newPost.put("lastName", lastName);
        newPost.put("phone", phone);
        newPost.put("gender", gender);
        newPost.put("id_number", id_number);
        newPost.put("email", email);

        progressDialog.get().setMessage(getResources().getString(R.string.Registering_Please_Wait));
        progressDialog.get().show();
        landlordRef.setValue(newPost);
        progressDialog.get().dismiss();

        Toast.makeText(LandlordRegistration.this, getResources().getString(R.string.landlord_added), Toast.LENGTH_LONG).show();

        Intent intent = new Intent(LandlordRegistration.this, RegisterLandlordUnits.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/stavros/demo/android/form/LandlordRegistration.kt
package com.stavros.demo.android.form

import android.app.ProgressDialog
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.stavros.demo.android.R
import java.util.*

class LandlordRegistration : AppCompatActivity() {

    private var uFirstName: EditText? = null
    private var uLastName: EditText? = null
    private var uPhoneNumber: EditText? = null
    private var uID_number: EditText? = null
    private var uEmail: EditText? = null
    private var radiog_Gender: RadioGroup? = null
    private var uGender: RadioButton? = null
    private var btn_save: Button? = null
    private var landlordRef: DatabaseReference? = null
    private val progressDialog = ThreadLocal<ProgressDialog>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_landlord_registration)
        uFirstName = findViewById(R.id.firstName)
        uLastName = findViewById(R.id.lasttName)
        uPhoneNumber = findViewById<View>(R.id.phoneNumber) as EditText
        uID_number = findViewById(R.id.id_number)
        uEmail = findViewById(R.id.email)
        radiog_Gender = findViewById(R.id.gender)
        btn_save = findViewById(R.id.btn_save)
        progressDialog.set(ProgressDialog(this))
        btn_save?.setOnClickListener { saveLandlord() }
    }

    private fun saveLandlord() {
        val firstName = uFirstName?.text.toString().trim { it <= ' ' }
        val lastName = uLastName?.text.toString().trim()
        val phone = uPhoneNumber?.text.toString().trim()
        val genderBtn = radiog_Gender?.checkedRadioButtonId
        var gender = ""
        uGender = findViewById(genderBtn!!)

        val id_number = uID_number?.text.toString().trim()
        val email = uEmail?.text.toString().trim()

        if (firstName.isEmpty()) {
            resources.getString(R.string.Please_enter_first_name).showLongToast()
            return
        }

        if (lastName.isEmpty()) {
            resources.getString(R.string.Please_enter_last_name).showLongToast()
            return
        }

        if (phone.isEmpty()) {
            resources.getString(R.string.Please_enter_phone_number).showLongToast()
            return
        } else if (phone.length != 10) {
            resources.getString(R.string.Please_enter_phone_number).showLongToast()
            return
        }

        gender = if (uGender != null) {
            uGender?.text.toString()
        } else {
            resources.getString(R.string.Please_select_gender).showLongToast()
            return
        }

        if (id_number.isEmpty()) {
            resources.getString(R.string.Please_enter_id_number).showLongToast()
            return
        }

        if (email.isEmpty()) {
            resources.getString(R.string.Please_enter_email).showLongToast()
            return
        }

        landlordRef = FirebaseDatabase.getInstance().reference.child("Users").child("Landlords").child(id_number)
        val newPost: MutableMap<String, Any?> = HashMap()
        newPost["firstName"] = firstName
        newPost["lastName"] = lastName
        newPost["phone"] = phone
        newPost["gender"] = gender
        newPost["id_number"] = id_number
        newPost["email"] = email

        progressDialog.get()?.setMessage(resources.getString(R.string.Registering_Please_Wait))
        progressDialog.get()?.show()

        landlordRef!!.setValue(newPost)
        progressDialog.get()?.dismiss()

        resources.getString(R.string.landlord_added).showLongToast()
        val intent = Intent(this@LandlordRegistration, RegisterLandlordUnits::class.java)
        startActivity(intent)
    }

    private fun String.showLongToast() {
        Toast.makeText(this@LandlordRegistration, this, Toast.LENGTH_LONG).show();
    }
}